src/Schema/SQLServerSchemaManager.php

```diff
@@ -227,9 +227,15 @@ protected function _getPortableTableForeignKeysList($tableForeignKeys)
             $name = $tableForeignKey['ForeignKey'];
 
             if (! isset($foreignKeys[$name])) {
+                $referencedTableName = $tableForeignKey['ReferenceTableName'];
+
+                if ($tableForeignKey['ReferenceSchemaName'] !== 'dbo') {
```

Comment:
We still have to default to the `dbo` schema here because the schema manager is unaware of the current schema in 3.x (see https://github.com/doctrine/dbal/pull/6677).